src/roswire/ros2/node.py

```diff
@@ -5,20 +5,34 @@
 
 import attr
 
+from ..interface import Node
+
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
     """Provides an interface for interacting with a ROS2 node.
 
     Attributes
     ----------
     name: str
         The name of the node.
     """
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    __name: str
+
+    @classmethod
+    def for_app_instance_and_name(cls,
+                                  name: str,
```

Comment:
I would swap `name` and `app_instance` to match the order specified in the method name.